src/signals/messages.nim

```diff
@@ -140,7 +140,12 @@ proc toTextItem*(jsonText: JsonNode): TextItem =
 
 
 proc toMessage*(jsonMsg: JsonNode): Message =
-  result = Message(
+  let
+    regex = re(r"(\b(https?|ftp):\/\/[-A-Z0-9+&@#\/%?=~_|!:,.;]*[-A-Z0-9+&@#\/%=~_|].(?:jpg|jpeg|gif|png|svg))", flags = {reStudy, reIgnoreCase})
```

Comment:
We already parse links in the text, you can get them from `parsedText` ("type" -> "link", "destination" -> the actual link), it's probably faster to iterate over it, pull the links and check the last n characters for the extension, instead of re-parsing the message with a regex (we don't parse links with regex for perf and security reasons).
We do not support ftp though, but I'd expect most will post http links.

Reply:
@cammellos very good point! I've added another commit that addresses this: https://github.com/status-im/nim-status-client/pull/500/commits/cf40c75d2052d3845f53b08a8a98ffb06e842144

Also think we're fine with not supporting `ftp` for the time being.